pkg/cloudclient/gcp/private.go

```diff
@@ -0,0 +1,376 @@
+package gcp
+
+//Features to add - image-id, kms-key-id
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"regexp"
+	"time"
+
+	"golang.org/x/oauth2/google"
+	computev1 "google.golang.org/api/compute/v1"
+
+	ocmlog "github.com/openshift-online/ocm-sdk-go/logging"
+	handledErrors "github.com/openshift/osd-network-verifier/pkg/errors"
+	"github.com/openshift/osd-network-verifier/pkg/helpers"
+	"github.com/openshift/osd-network-verifier/pkg/output"
+)
+
+type createComputeServiceInstanceInput struct {
+	ContOptImageID string
+	vpcSubnetID    string
+	userdata       string
+	zone           string
+	machineType    string
+	instanceName   string
+	sourceImage    string
+	networkName    string
+}
+
+var (
+	// TODO find a location for future docker images
+	networkValidatorImage string = "quay.io/app-sre/osd-network-verifier:v0.1.159-9a6e0eb"
+	userdataEndVerifier   string = "USERDATA END"
+)
+
+func newClient(ctx context.Context, logger ocmlog.Logger, credentials *google.Credentials, region, instanceType string, tags map[string]string) (*Client, error) {
+	//use oauth2 token in credentials struct to create a client,
+	// https://pkg.go.dev/golang.org/x/oauth2/google#Credentials
+
+	// https://cloud.google.com/docs/authentication/production
+	//service account credentials order/priority - env variable, service account attached to resource, error
+
+	computeService, err := computev1.NewService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Client{
+		projectID:      credentials.ProjectID,
+		region:         region,
+		zone:           fmt.Sprintf("%s-b", region), //append zone b
```

Comment:
is b arbitrarily chosen? if so, it maybe worth specifying in the comment

Reply:
Zone b is supported by all regions and has the most machine types compared to zone a and c. Zone a doesn't exist for regions europe-west1 and us-east1. Zone c is also supported by all regions but lacks machine types compared to zone b. Docs: https://cloud.google.com/compute/docs/regions-zones#available

Reply:
would you mind adding that rationale to the comment
`//appending zone b, since it is available in all regions (see https://cloud.google.com/compute/docs/regions-zones#available)`